ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactAnalyzer.java

```diff
@@ -80,13 +91,30 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
         case HiveParser.TOK_ORDERBY:
           orderBy = this.ctx.getTokenRewriteStream().toOriginalString(node.getTokenStartIndex(), node.getTokenStopIndex());
           break;
+        case HiveParser.TOK_WHERE:
+          RowResolver rwsch = new RowResolver();
+          Map<String, String> colTypes = new HashMap<>();
+          try {
+            for (FieldSchema fs : getDb().getTable(tableName).getCols()) {
+              TypeInfo columnType = TypeInfoUtils.getTypeInfoFromTypeString(fs.getType());
+              rwsch.put(getDb().getTable(tableName).getTableName(), fs.getName(), 
```

Comment:
why are we getting tableName here from getDb().getTable(tableName).getTableName(), even though tableName variable itself is present?

Reply:
Good catch, done.